x-pack/platform/plugins/shared/dataset_quality/public/utils/generate_datasets.ts

```diff
@@ -74,6 +74,7 @@ export function generateDatasets(
     {}
   );
 
+  const datasetsWithFailureStore = new Set(failedDocStats.map(({ dataset }) => dataset));
```

Comment:
How are we getting the `hasFailureStore` from here? afaik it's coming in the dataStreamStats rather than `failedDocStats`